backend/routes/userRoutes.js

```diff
@@ -105,4 +105,142 @@ router.patch('/location', async(req, res) => {
   res.json(updatedUser)
 })
 
+router.get('/recommendedEvents/:userId', async (req, res) => {
+  const { userId } = req.params
+  let idRecommendedEvents = [];
+  let scoreRecommendedEvents = [];
+  let categoriesEventsAttending = [];
+  let idEventsAttending = [];
+  let totalDistanceUserToEvent = 0;
+
+  // Get the current user
+  const user = await prisma.user.findFirst({
+    where: { id: parseInt(userId) }
+  })
+  // ----------------------------------------
+  // Get the events that our user is attending
+  const attendance = await prisma.attendance.findMany({
+    where: { 
+      userAttending: user.username
+    }
+  })
+  attendance.map((d) => 
+    idEventsAttending.push(d.eventId)
+  )
+
+  const eventsAttending = await prisma.event.findMany({
+    where: {
+      id: { in: idEventsAttending}
+    }
+  })
+  // --------------------------------------------
+  // Get the categories of the events that the user is attending
+  eventsAttending.map((e) => 
+    categoriesEventsAttending.push(e.category)
+  )
+  // ----------------------------------------------
+  // Get the mean distance that our user is willing to travel for an event
+  eventsAttending.map((e) => 
+    totalDistanceUserToEvent = ( ((e.lat - user.lat)**2 + (e.long - user.long)**2)**.5 ) + totalDistanceUserToEvent
+  )
+  let numberOfEventsAttending = Object.keys(eventsAttending).length
+  let meanDistanceTraveledUser = totalDistanceUserToEvent/numberOfEventsAttending
+  // -------------------------------------------------------------------------------
+  // Get the events that are nearby (square with side = 28 miles)
+  const eventsNear = await prisma.event.findMany({
+    where: { 
+      lat: {
+        gt: user.lat - 0.2,
+        lt: user.lat + 0.2
+      },
+      long: {
+        gt: user.long - 0.2, 
+        lt: user.long + 0.2 
+      }
+    }
+  })
```

Comment:
To push this further, can we make this customized to the user? For some people, 5 miles is the furthest, and for some people 100 miles is acceptable. Since you already have history of user's attendance, can you derive anything from there?

Reply:
Yes, will solve in todays PR

Reply:
Solved in PR 39 by taking in account the average distance a user travels for an event